server_test.go

```diff
@@ -1,6 +1,112 @@
 package main
 
-import "testing"
+import (
+	"testing"
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"net"
+	"time"
+	"github.com/stretchr/testify/require"
+	"encoding/hex"
+	"github.com/lightningnetwork/lnd/brontide"
+	"github.com/roasbeef/btcd/connmgr"
+	"fmt"
+	"strings"
+	"io"
+)
+
+var (
+	privKey = createPrivateKey()
+	pubKey = createPubKey()
+	compressedPubKey = string(pubKey.SerializeCompressed())
+	address = lnwire.NetAddress{IdentityKey: pubKey, Address: &StubAddress{}}
+)
+
+func TestConnectToPeer_NonPerm(t *testing.T) {
+	srv := createServer()
+	err := srv.ConnectToPeer(&address, false)
+
+	require.NoError(t, err)
+
+	require.NotNil(t, srv.peersByPub[compressedPubKey], "Peer not added")
+	require.NotNil(t, srv.inboundPeers[compressedPubKey], "Peer not added")
+
+	require.Equal(t, 0, len(srv.persistentPeers),
+		"No persistent peers should be present.")
+}
+
+func TestConnectToPeer_Perm(t *testing.T) {
+	srv := createServer()
+	err := srv.ConnectToPeer(&address, true)
+
+	require.NoError(t, err)
+
+	// Interestingly, when perm is specified, the peer does NOT
```

Comment:
This is incorrect, it _will_ end up in those maps. The diff is that it's done asynchronously, since we'll launch a `cmgr` connection to retry as needed.